tests/test_search.py

```diff
@@ -265,7 +271,7 @@ def test_IndexBuilder():
     index.dump(stream, 'pickle')
     stream.seek(0)
 
-    index2 = IndexBuilder(env, 'en', {}, None)
+    index2 = IndexBuilder(env, 'en', {}, '')
```

Comment:
I'd be tempted to suggest that we make the `scoring` option to `IndexBuilder` into an optional string -- or even an optional string with default `None`, so that this change isn't necessary.

Reply:
Why do you want to avoid this change?

Reply:
It seems we have a choice between either changing the type-documented interface of the `IndexBuilder` -- in a way that allows it to accept precisely one more valid value (the value `None`), for which there already exists test coverage -- or to adjust the test case to fit within the currently-documented interface.

Either seems reasonable -- but to me, if I don't want to add a custom `scorer` when I'm using an `IndexBuilder`, then I'd prefer to indicate that by passing a `None` value - "I don't have a custom scorer to provide", rather than saying "here's an empty string that may or may not be my custom scorer" (currently it isn't considered a custom scorer; it's filtered-out by a conditional branch in the implementation).

In addition I _think_ that when a `sphinx.config.Config` value is not set, it takes the value `None`, so there's arguably some consistency there (but I have to admit to not checking that yet, I've only inspected the code).

Reply:
Oh, hold on: it's a filename argument, not the literal scorer code.. so an empty-string filename is used to communicate that we should use no custom scorer.  I'll read a bit more of this code to figure out what's going on here.

Reply:
> Oh, hold on: it's a filename argument, not the literal scorer code.. so an empty-string filename is used to communicate that we should use no custom scorer. I'll read a bit more of this code to figure out what's going on here.

The argument is essentially a file path (as a `str`) -- and when it is falsy (either `None` or an empty string), no custom search scoring code is loaded.  I think that aligns reasonably with adjusting the type-hint to `str | None` for now (although potentially becoming optional-`path` in future), as in the PR #12829 I've opened.

Reply:
So to answer the original question: I'd prefer that we fix the type-hint (a change that doesn't affect application runtime behaviour) instead of changing the test coverage (unless there's a good reason to do that).

Reply:
@jayaddison what's the current state of the art on this? I know you proposed some changes- did they get merged?

Reply:
Hi @danieleades - nope, those are still pending in https://github.com/sphinx-doc/sphinx/pull/12829 at the moment.  What do you think about the suggested change in that pull request?

Reply:
i don't have any strong feelings to be honest, so long as the implementation and the type annotations are consistent i can go about my merry, pedantic way

Reply:
It should be `''`, see https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_search_scorer